libs/server-sent-events/src/client.cpp

```diff
@@ -79,41 +79,48 @@ class FoxyClient : public Client,
           read_timeout_(read_timeout),
           write_timeout_(write_timeout),
           req_(std::move(req)),
-          session_(std::move(executor),
-                   launchdarkly::foxy::session_opts{
-                       ToOptRef(ssl_context_),
-                       connect_timeout.value_or(kNoTimeout)}),
+          event_receiver_(std::move(receiver)),
+          logger_(std::move(logger)),
+          errors_(std::move(errors)),
+          body_parser_(std::nullopt),
+          session_(std::nullopt),
+          last_event_id_(std::nullopt),
           backoff_(
               initial_reconnect_delay.value_or(kDefaultInitialReconnectDelay),
               kDefaultMaxBackoffDelay),
-          last_event_id_(std::nullopt),
-          backoff_timer_(session_.get_executor()),
-          event_receiver_(std::move(receiver)),
-          logger_(std::move(logger)),
-          errors_(std::move(errors)) {
+          backoff_timer_(std::move(executor)),
+          last_read_(std::nullopt),
+          shutting_down_(false) {
```

Comment:
Session construction now takes place in `create_session()`, so the logic can be reused for each new connection.